From ee11b85ed52b396e962a5819a11542b9b00f24e0 Mon Sep 17 00:00:00 2001
From: Jan Sondhauss <jan.sondhauss@wago.com>
Date: Wed, 3 Jun 2020 14:04:44 +0200
Subject: [PATCH] init_post_autoboot: pfc-switch-reset: add sleep after getting
 the switch out of reset

A generous second should be sufficient to ensure that the switch is ready
to respond. This delay is necessary because of the changed order in which
the init scripts are executed. Previously the pfc-switch-reset script
was executed at boot-stage. Due to the move to init_post_boot, the time
period between the reset and the first access to the switch is
too short in order to guarantee that the switch is able to respond on
the mdio bus.

Co-authored-by: Oleg Karfich <oleg.karfich@wago.com>
Signed-off-by: Jan Sondhauss <jan.sondhauss@wago.com>
---
 defaultenv/defaultenv-pfc-base/init_post_autoboot/a-pfc-switch-reset | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/defaultenv/defaultenv-pfc-base/init_post_autoboot/a-pfc-switch-reset b/defaultenv/defaultenv-pfc-base/init_post_autoboot/a-pfc-switch-reset
index f2d35d5..46e08fb 100644
--- a/defaultenv/defaultenv-pfc-base/init_post_autoboot/a-pfc-switch-reset
+++ b/defaultenv/defaultenv-pfc-base/init_post_autoboot/a-pfc-switch-reset
@@ -20,3 +20,5 @@ fi
 gpio_direction_output "${global.pfc.sw_reset_gpio}" 0
 msleep 1
 gpio_direction_output "${global.pfc.sw_reset_gpio}" 1
+# give the switch and phys some time to wakeup from reset
+msleep 1000
-- 
2.7.4

